aten/src/ATen/native/cpu/Loops.h

```diff
@@ -127,6 +127,63 @@ basic_loop(char* C10_RESTRICT data[], const int64_t* strides_, int64_t i, int64_
   execute_op(data, strides, i, n, std::forward<func_t>(op));
 }
 
+template<class T, size_t N>
+struct TupleOutput {
+  static void handle(char *C10_RESTRICT data[], const int64_t *strides, int64_t i,
+                     const T &tuple) {
+    TupleOutput<T, N - 1>::handle(data, strides, i, tuple);
+
+    auto output = std::get<N - 1>(tuple);
+    using output_type = decltype(output);
+    output_type * out_ptr = (output_type *)(data[N - 1] + i * strides[N - 1]);
+    *out_ptr = output;
+  }
+};
+
+template<class T>
+struct TupleOutput<T, 1> {
+  static void handle(char *C10_RESTRICT data[], const int64_t *strides, int64_t i,
+                     const T &tuple) {
+    auto output = std::get<0>(tuple);
+    using output_type = decltype(output);
+    output_type* out_ptr = (output_type *)(data[0] + i * strides[0]);
+    *out_ptr = output;
+  }
+};
+
+template<class... Args>
+void handle_tuple_outputs(char* C10_RESTRICT data[],
+                          const int64_t* strides,
+                          int64_t i,
+                          const std::tuple<Args...> &tuple) {
+  TupleOutput<decltype(tuple), sizeof...(Args)>::handle(data, strides, i, tuple);
+}
+
+template <typename func_t>
+static inline void
+multiple_outputs_loop(char* C10_RESTRICT data[], const int64_t* strides_, int64_t i, int64_t n, func_t&& op) {
```

Comment:
It can handle multiple inputs.
This PR also updates `aten/src/ATen/test/tensor_iterator_test.cpp` by adding a binary ops test case for testing the functionality of `cpu_kernel_multiple_outputs`.
```c++
auto iter = at::TensorIteratorConfig()                                                            \
  .add_output(out1)                                                                               \
  .add_output(out2)                                                                               \
  .add_input(in1)                                                                                 \
  .add_input(in2)                                                                                 \
  .build();                                                                                       \
at::native::cpu_kernel_multiple_outputs(iter, [=](ctype a, ctype b) -> std::tuple<ctype, ctype> { \
  ctype add = a + b;                                                                              \
  ctype mul = a * b;                                                                              \
  return std::tuple<ctype, ctype>(add, mul);                                                      \
}); 
```